<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1057</width>
    <height>642</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenGLDemo</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="MyGL" name="mygl">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>11</y>
      <width>618</width>
      <height>432</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="vertsListWidget">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>30</y>
      <width>111</width>
      <height>201</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="halfEdgesListWidget">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>30</y>
      <width>111</width>
      <height>201</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="facesListWidget">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>30</y>
      <width>111</width>
      <height>201</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>10</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertices</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>10</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Half-Edges</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>10</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Faces</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="vertPosXSpinBox">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>240</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.989999999999995</double>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="vertPosYSpinBox">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>240</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.989999999999995</double>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="vertPosZSpinBox">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>240</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.989999999999995</double>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>240</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertex Position</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>260</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>260</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>260</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>310</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Blue</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>310</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Red</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="faceBlueSpinBox">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>290</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.050000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>290</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Face Color</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="faceGreenSpinBox">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>290</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.050000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="faceRedSpinBox">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>290</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.050000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>310</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Green</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="ClearButton">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>330</y>
      <width>91</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear Selected</string>
    </property>
   </widget>
   <widget class="QPushButton" name="AddVertexButton">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>240</y>
      <width>80</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Add Vertex</string>
    </property>
   </widget>
   <widget class="QPushButton" name="TriangulateButton">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>280</y>
      <width>80</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Triangulate</string>
    </property>
   </widget>
   <widget class="QPushButton" name="SubdivisionButton">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>320</y>
      <width>80</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Subdivision</string>
    </property>
   </widget>
   <widget class="QLabel" name="skeletonLabel">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>360</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Skeleton</string>
    </property>
   </widget>
   <widget class="QTreeWidget" name="jointsTreeWidget">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>380</y>
      <width>401</width>
      <height>231</height>
     </rect>
    </property>
    <column>
     <property name="text">
      <string/>
     </property>
    </column>
   </widget>
   <widget class="QPushButton" name="BindSkinButton">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>450</y>
      <width>80</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Bind Skin</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>480</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Joint Translation</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>530</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Joint Rotation</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>460</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>460</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>460</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="jointPosXAddButton">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>480</y>
      <width>31</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>+</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointPosXSubButton">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>480</y>
      <width>31</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointPosYSubButton">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>480</y>
      <width>31</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointPosYAddButton">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>480</y>
      <width>31</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>+</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointPosZSubButton">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>480</y>
      <width>31</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointPosZAddButton">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>480</y>
      <width>31</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>+</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointRotZAddButton">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>530</y>
      <width>31</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>+</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointRotXSubButton">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>530</y>
      <width>31</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointRotYAddButton">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>530</y>
      <width>31</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>+</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointRotXAddButton">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>530</y>
      <width>31</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>+</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointRotYSubButton">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>530</y>
      <width>31</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointRotZSubButton">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>530</y>
      <width>31</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1057</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionImport_OBJ"/>
    <addaction name="actionImport_JSON"/>
    <addaction name="actionQuit"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionCamera_Controls"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionCamera_Controls">
   <property name="text">
    <string>Camera Controls</string>
   </property>
  </action>
  <action name="actionImport_OBJ">
   <property name="text">
    <string>Import OBJ</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionImport_JSON">
   <property name="text">
    <string>Import JSON</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+J</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGL</class>
   <extends>QOpenGLWidget</extends>
   <header>mygl.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
